src/sentry/static/sentry/app/views/eventsV2/eventView.tsx

```diff
@@ -218,19 +282,49 @@ class EventView {
 
     return new EventView({
       fields,
+      name: eventViewV1.name,
       sorts: fromSorts(eventViewV1.data.sort),
       tags: eventViewV1.tags,
       query: eventViewV1.data.query,
+      project: [],
+      range: undefined,
+      start: undefined,
+      end: undefined,
+    });
+  }
+
+  static fromSavedQuery(saved: SavedQuery): EventView {
+    const fields = saved.fields.map(field => {
+      return {field, title: field};
+    });
+
+    return new EventView({
+      fields,
+      name: saved.name,
+      query: queryStringFromSavedQuery(saved),
+      project: saved.projects,
+      start: saved.start,
+      end: saved.end,
+      range: saved.range,
+      sorts: [],
+      tags: [],
     });
   }
 
   generateQueryStringObject(): Query {
-    return {
+    const output = {
       field: encodeFields(this.fields),
       sort: encodeSorts(this.sorts),
       tag: this.tags,
       query: this.query,
     };
+    const conditionalFields = ['name', 'project', 'start', 'end', 'range'];
+    for (const field of conditionalFields) {
+      if (this[field] && this[field].length) {
+        output[field] = this[field];
```

Comment:
Is there a way we can make sure consumers of the output don't try to mutate it, would it be worth cloning the array?

Reply:
We could use `_.cloneDeep` to give a deep copy out. That would avoid mutations from coming back to the EventView.

Reply:
Wonder if we could also just handle this in typescript? e.g. https://spin.atomicobject.com/2017/08/14/typescript-readonly-intro/